distrib/setup.sh

```diff
@@ -0,0 +1,123 @@
+  ____) |  | |/ ____ \| | \ \| |____ / ____ \| . \| |____
+ |_____/   |_/_/    \_\_|  \_\______/_/    \_\_|\_\______|
+
+
+EOF
+}
+
+get_installation_directory() {
+    read -p "Where do you want to install Starlake? [$HOME/starlake]: " INSTALL_DIR
+    INSTALL_DIR=${INSTALL_DIR:-$HOME/starlake}
+    mkdir -p "$INSTALL_DIR"
+}
+
+get_from_url() {
+    local url=$1
+    echo "Curling $url"
+    local response=$(curl -s -w "%{http_code}" "$url")
+    local status_code=${response: -3}
+
+    if [[ ! $status_code =~ ^(2|3)[0-9][0-9]$ ]]; then
+        echo "Error: Failed to retrieve data from $url. HTTP status code: $status_code"
+        exit 1
+    fi
+
+    # Print the content excluding the status code
+    local content_length=${#response}
+    local content="${response:0:content_length-3}"
+    echo "$content"
+}
+
+get_version_to_install() {
+    # Extract the version number from command-line arguments
+    for arg in "$@"; do
+        if [[ $arg == "--version="* ]]; then
+            VERSION="${arg#*=}"
+        fi
+    done
+
+    ALL_SNAPSHOT_VERSIONS=$(get_from_url https://s01.oss.sonatype.org/service/local/repositories/snapshots/content/ai/starlake/starlake-spark3_2.12/ | awk -F'<|>' '/<text>/{print $3}' | grep -oE '^[0-9]+\.[0-9]+\.[0-9]+-SNAPSHOT$' | sort -rV)
+    ALL_RELEASE_VERSIONS=$(get_from_url https://s01.oss.sonatype.org/service/local/repositories/releases/content/ai/starlake/starlake-spark3_2.12/ | awk -F'<|>' '/<text>/{print $3}' | grep -oE '^[0-9]+\.[0-9]+\.[0-9]+$' | sort -rV)
+
+    SNAPSHOT_VERSION=$(echo "$ALL_SNAPSHOT_VERSIONS" | head -n 1)
+    LATEST_RELEASE_VERSIONS=$(echo "$ALL_RELEASE_VERSIONS" | head -n 5)
+
+    VERSIONS=("$SNAPSHOT_VERSION" $LATEST_RELEASE_VERSIONS)
+    VERSIONS=$VERSIONS
+
+     while [[ ! "$VERSIONS" =~ (^|[[:space:]])"$VERSION"($|[[:space:]]) ]]; do
+        echo "Invalid version $VERSION. Please choose from the available versions."
+        echo "Last 5 available versions:"
+        for version in "${VERSIONS[@]}"; do
+            echo "$version"
+        done
+        read -p "Which version do you want to install? [$(echo "$VERSIONS" | head -n 1)]: " VERSION
+        VERSION=${VERSION:-$(echo "$VERSIONS" | head -n 1)}
+     done
+
+}
+
+install_starlake() {
+    echo "installing $VERSION"
+    if [[ $VERSION == *"SNAPSHOT"* ]]; then
+        local url=https://raw.githubusercontent.com/starlake-ai/starlake/master/distrib/starlake.sh
+    else
+        local url=https://raw.githubusercontent.com/starlake-ai/starlake/v$VERSION/distrib/starlake.sh
+    fi
+
+    get_from_url $url > "$INSTALL_DIR/starlake"
+
+    chmod +x "$INSTALL_DIR/starlake"
+}
+
+
+add_starlake_to_path() {
+    if [ -f "$HOME/.zshrc" ] || [ -f "$HOME/.bashrc" ]; then
+        if [ -f "$HOME/.zshrc" ]; then
+            if ! grep -q "$INSTALL_DIR" ~/.zshrc; then
+                echo -e "\nexport PATH=$INSTALL_DIR:\$PATH" >> ~/.zshrc
+            fi
+            zsh ~/.zshrc
```

Comment:
```
if [[ `echo $0` == *zsh* ]]; then  zsh ~/.zshrc; fi
```

Reply:
Interesting, i didn't know it exists.
I test it in the terminal it works, but wehn i a dd it in the bash it does not work. I ended up using if [[ "$SHELL" == *zsh* ]], WDYT ?